### PR TITLE
[BugFix] Fix set variable = default NPE bug and Refactor Variable analyzer

### LINKED ISSUE #17562: Set variable = default at follower FE cause NPE
### Steps to reproduce the behavior (Required)
1、Connect to FE follower
2、 set query_mem_limit=DEFAULT;
3、set global query_mem_limit=DEFAULT;

### Expected behavior (Required)
success
### Real behavior (Required)
ERROR 1064 (HY000): Unknown error
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by HangyuanLiu**
SET variable=DEFAULT, Expr in SetVar will be saved as NULL. This NULL value will be treated specially in the VariableMgr#setVar function. So no problem on FE master. But when the command that needs to forward to the master is run on the FE follower, AST2SQL in `params.setModified_variables_sql(AstToSQLBuilder.toSQL(setStmt));` does not specially handle this NULL expression, resulting in NPE

Signed-off-by: HangyuanLiu <460660596@qq.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #17562

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 255 / 317 (80.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|0|10|00.00%|[111, 117, 118, 119, 120, 123, 124, 126, 127, 128]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowVariablesStmt.java|0|1|00.00%|[81]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[365, 366]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|20|00.00%|[363, 364, 365, 367, 368, 369, 370, 374, 376, 378, 379, 380, 382, 383, 385, 389, 391, 393, 403, 404]|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|0|7|00.00%|[501, 502, 503, 506, 507, 508, 510]|
|:large_blue_circle:|com/starrocks/qe/SetExecutor.java|9|12|75.00%|[85, 140, 149]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|6|7|85.71%|[1460]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetStmtAnalyzer.java|127|142|89.44%|[69, 89, 101, 102, 107, 108, 109, 149, 188, 198, 199, 249, 253, 259, 275]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|29|31|93.55%|[2924, 2928]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToStringBuilder.java|22|23|95.65%|[178]|
|:large_blue_circle:|com/starrocks/sql/ast/SetNamesVar.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetStmt.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UserVariable.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetPassVar.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetTransaction.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetUserPropertyStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SystemVariable.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetListItem.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetType.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SetStmtAnalyzer.java [hunk 1]
```diff
@@ -14,76 +14,286 @@
...
+
+    private static void analyzeSetPassVar(SetPassVar var, ConnectContext session) {
+        try {
+            UserIdentity userIdentity = var.getUserIdent();
+            if (userIdentity == null) {
+                userIdentity = session.getCurrentUserIdentity();
+            }
+            userIdentity.analyze();
+            var.setUserIdent(userIdentity);
+            var.setPasswdBytes(MysqlPassword.checkPassword(var.getPasswdParam()));
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SetStmtAnalyzer.java:293]: I see some auth process in the old code(`SetPassVar::analyze`). It is no longer needed?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SetStmtAnalyzer.java:293]: Yes, it's no longer needed, the new version's permission logic has been moved into PrivilegeCheckerV2

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Review by liuyehcf — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 250 / 310 (80.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|0|10|00.00%|[111, 117, 118, 119, 120, 123, 124, 126, 127, 128]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowVariablesStmt.java|0|1|00.00%|[81]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[369, 370]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|20|00.00%|[363, 364, 365, 367, 368, 369, 370, 374, 376, 378, 379, 380, 382, 383, 385, 389, 391, 393, 403, 404]|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|0|7|00.00%|[501, 502, 503, 506, 507, 508, 510]|
|:large_blue_circle:|com/starrocks/qe/SetExecutor.java|9|12|75.00%|[85, 140, 149]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|6|7|85.71%|[1460]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetStmtAnalyzer.java|127|142|89.44%|[69, 89, 101, 102, 107, 108, 109, 149, 188, 198, 199, 249, 253, 259, 275]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToStringBuilder.java|21|22|95.45%|[178]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|25|25|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetNamesVar.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetStmt.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UserVariable.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetPassVar.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetTransaction.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetUserPropertyStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SystemVariable.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetListItem.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetType.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|[]|

**Pull request merged**